src/base.c

```diff
@@ -377,10 +377,11 @@ int cw_authenticate(const char *service, const char *username, const char *passw
 	};
 
 	rc = pam_start(service, creds.username, &convo, &pam);
-	if (rc == PAM_SUCCESS)
+	if (rc == PAM_SUCCESS) {
 		rc = pam_authenticate(pam, PAM_DISALLOW_NULL_AUTHTOK);
 		if (rc == PAM_SUCCESS)
 			rc = pam_acct_mgmt(pam, PAM_DISALLOW_NULL_AUTHTOK);
+	}
```

Comment:
I think we want to bail on account management if we fail to start, or fail to auth. In essence, if we fail on the previous pam step, then we probably don't want to try the next.